internal/pkg/bulk/engine.go

```diff
@@ -65,18 +68,28 @@ type Bulk interface {
 	// Accessor used to talk to elastic search direcly bypassing bulk engine
 	Client() *elasticsearch.Client
 
+	CheckRemoteOutputChanged(name string, newCfg map[string]interface{})
+
+	RemoteOutputCh() chan bool
+
+	CreateAndGetBulker(outputName string, serviceToken string, outputMap map[string]map[string]interface{}) (Bulk, error)
+	GetBulker(outputName string) Bulk
+
 	ReadSecrets(ctx context.Context, secretIds []string) (map[string]string, error)
 }
 
 const kModBulk = "bulk"
 
 type Bulker struct {
-	es          esapi.Transport
-	ch          chan *bulkT
-	opts        bulkOptT
-	blkPool     sync.Pool
-	apikeyLimit *semaphore.Weighted
-	tracer      *apm.Tracer
+	es                    esapi.Transport
+	ch                    chan *bulkT
+	opts                  bulkOptT
+	blkPool               sync.Pool
+	apikeyLimit           *semaphore.Weighted
+	tracer                *apm.Tracer
+	remoteOutputConfigMap map[string]map[string]interface{}
+	remoteOutputCh        chan bool
+	bulkerMap             map[string]Bulk
```

Comment:
moved the remove bulker map to the bulker itself (out of policy_output) so it can be called from handleAck to invalidate api keys